internal/s3shared/resource_request.go

```diff
@@ -26,6 +26,8 @@ func (r ResourceRequest) AllowCrossRegion() bool {
 }
 
 // UseFIPS returns true if request config region is FIPS
+//
+// Deprecated: FIPS pseudo-regions are deprecated and are translated to appropriate endpoint options.
```

Comment:
These `UseFIPS` and `IsFIPS` helpers can be deleted because the v1 SDK is a mono-module, so don't have to worry about old downstream dependencies not being compatible with the change.